demo_drivers/meshing/mesh_from_gmsh/mesh_from_gmsh_solid.cc

```diff
@@ -0,0 +1,273 @@
+            mesh_pt()->boundary_node_pt(b,i)->pin_position(1);
+            mesh_pt()->boundary_node_pt(b,i)->pin_position(2);
+        }
+
+        // Pin the redundant solid pressures (if any)
+        PVDEquationsBase<3>::pin_redundant_nodal_solid_pressures(mesh_pt()->element_pt());
+
+        //Assign equation numbers
+        assign_eqn_numbers();
+
+        // Prepare output directory
+        Doc_info.set_directory("RESLT");
+
+    }
+
+    /// Update function (empty)
+    void actions_after_newton_solve() {}
+
+    /// Update function (empty)
+    void actions_before_newton_solve() {}
+
+    /// Actions before adapt. Empty
+    void actions_before_adapt(){}
+
+    /// Actions after adapt
+    void actions_after_adapt()
+    {
+        // Pin the redundant solid pressures (if any)
+        PVDEquationsBase<3>::pin_redundant_nodal_solid_pressures(mesh_pt()->element_pt());
+    }
+
+    /// Doc the solution
+    void doc_solution()
+    {
+
+        ofstream some_file;
+        char filename[100];
+
+        // Number of plot points
+        unsigned n_plot = 2;
+
+        // Output shape of deformed body
+        sprintf(filename,"%s/soln%i.dat",Doc_info.directory().c_str(),Doc_info.number());
+        some_file.open(filename);
+        mesh_pt()->output(some_file,n_plot);
+        some_file.close();
+
+        // Increment label for output files
+        Doc_info.number()++;
+
+    };
+
+
+
+    /// Access function for the mesh
+    SolidGmshBrickMesh<ELEMENT>* mesh_pt()
+    {
+        return dynamic_cast<SolidGmshBrickMesh<ELEMENT>*>(Problem::mesh_pt());
+    }
+
+
+    void solve()
+    {
+
+        // Set output directory
+        char dirname[100];
+
+        sprintf(dirname,"RESLT");
+
+
+        // Prepare output
+        Doc_info.set_directory(dirname);
+
+        // Doc solution
+        doc_solution();
+
+
+        //Parameter incrementation
+        unsigned nstep=5;
+
+        double g_increment=1.0e-1;
+        for(unsigned i=0;i<nstep;i++)
+        {
+            // Increment load
+            Input::Gravity+=g_increment;
+
+
+            // Solve it
+            newton_solve();
+
+
+            // Doc solution
+            doc_solution();
+
+        }
+
+    };
+
+private:
+
+    /// DocInfo object for output
+    DocInfo Doc_info;
+    SolidGmshBrickMesh<ELEMENT>* Solid_mesh_pt;
+};
+
+
+
+
+//=======start_of_main==================================================
+/// Driver for 3D cantilever beam loaded by gravity
+//======================================================================
+int main(int argc, char* argv[])
+{
+    // Store command line arguments
+    CommandLineArgs::setup(argc,argv);
+
+    // Check number of command line arguments: Need exactly three.
+    if (argc!=2)
+    {
+        std::string error_message =
+            "Wrong number of command line arguments.\n";
+        error_message +=
+            "Must specify the following file name  \n";
+        error_message += 
```

Comment:
Really? If so, we may as well hard-code it and not pass it in through the command line args. I would have thought that the code works for "any" (well, within limits due to the boundary enumeration) mesh file.

Reply:
I think it is better to have the program executed as ` ./myApplication mesh.msh` this facilitate testing different mesh files (with different configurations) and of course the user can hard code it in the driver code as you mentioned.